api/src/main/java/org/ehcache/spi/loaderwriter/CacheLoaderWriter.java

```diff
@@ -14,26 +14,47 @@
  * limitations under the License.
  */
 
-package org.ehcache.spi.writer;
+package org.ehcache.spi.loaderwriter;
 
 import java.util.Map;
-
-import org.ehcache.exceptions.BulkCacheWriterException;
+import org.ehcache.exceptions.BulkCacheWritingException;
 
 /**
- * A CacheWriter is associated with a given {@link org.ehcache.Cache} instance and will be used to maintain it in sync
- * with an underlying system of record during normal operations on the Cache.
+ * A CacheLoaderWriter is associated with a given {@link org.ehcache.Cache} instance and will be used to keep it
+ * in sync with another system.
  * <p>
  * Instances of this class have to be thread safe.
  * <p>
  * Any {@link java.lang.Exception} thrown by methods of this interface will be wrapped into a
- * {@link org.ehcache.exceptions.CacheWriterException} by the {@link org.ehcache.Cache} and will need to be handled by
+ * {@link org.ehcache.exceptions.CacheLoadingException} by the {@link org.ehcache.Cache} and will need to be handled by
```

Comment:
This javadoc seems to be a little misleading. Exceptions on the load methods will throw CacheLoadingException, but the write methods still throw CacheWriterException (or subclass) on Exception, no?